unbxdAutosuggest.js

```diff
@@ -621,91 +765,251 @@ var unbxdAutoSuggestFunction = function($,Handlebars,undefined){
 				,IN_FIELD : []
 			}
 			,infieldsCount = 0;
-
+			var key_count = 0;
 			for(var x = 0; x < data.response.products.length; x++){
 				var doc = data.response.products[x]
 					,o = {};
-				if("TOP_SEARCH_QUERIES" == doc.doctype && this.options.topQueries.count > this.currentResults.TOP_SEARCH_QUERIES.length){
-					o = {
-						autosuggest : doc.autosuggest
-						,highlighted : this.highlightStr(doc.autosuggest)
-						,type : "TOP_SEARCH_QUERIES"
-						,_original : doc.doctype
-					};
-					this.currentResults.TOP_SEARCH_QUERIES.push(o);
-				}else if("IN_FIELD" == doc.doctype && this.options.inFields.count > infieldsCount){
-					var ins = {}
-						,asrc = " " + doc.unbxdAutosuggestSrc + " "
-						,highlightedtext = this.highlightStr(doc.autosuggest);
-
-					for(var a in this.options.inFields.fields){
-						if( (a+"_in") in doc && doc[a+"_in"].length && asrc.indexOf(" " +a+" ") == -1){
-							ins[a] = doc[a+"_in"].slice(0, parseInt(this.options.inFields.fields[a]))
+				if(this.options.maxSuggestions){
+					if("TOP_SEARCH_QUERIES" == doc.doctype && count['topquery'] > this.currentResults.TOP_SEARCH_QUERIES.length ){
+						o = {
+							autosuggest : doc.autosuggest
+							,highlighted : this.highlightStr(doc.autosuggest)
+							,type : "TOP_SEARCH_QUERIES"
+							,_original : doc.doctype
+						};
+						this.currentResults.TOP_SEARCH_QUERIES.push(o);
+					}else if("IN_FIELD" == doc.doctype && (count['infields']+count['key_rem']+count['top_rem']) > infieldsCount ){
+						if(count['infields'] > infieldsCount){
+							var ins = {}
+							,asrc = " " + doc.unbxdAutosuggestSrc + " "
+							,highlightedtext = this.highlightStr(doc.autosuggest);
+
+							for(var a in this.options.inFields.fields){
+								if( (a+"_in") in doc && doc[a+"_in"].length && asrc.indexOf(" " +a+" ") == -1){
+									ins[a] = doc[a+"_in"].slice(0, parseInt(this.options.inFields.fields[a]))
+								}
+							}
+							if(doc.autosuggest.indexOf(doc.category_in[0])==-1){
```

Comment:
Can you explain what are you doing here  ? 
what if category_in is empty array ?